tests/script-based-pre/verify_std_cmd/verify_std.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+# Copyright Kani Contributors
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+set +e
+
+TMP_DIR="tmp_dir"
+
+rm -rf ${TMP_DIR}
+mkdir ${TMP_DIR}
+
+# Create a custom standard library.
+SYSROOT=$(rustc --print sysroot)
+STD_PATH="${SYSROOT}/lib/rustlib/src/rust/library"
+cp -r "${STD_PATH}" "${TMP_DIR}"
+
+# Insert a small harness in one of the standard library modules.
+CORE_CODE='
+#[cfg(kani)]
+#[unstable(feature = "kani", issue = "none")]
+pub mod kani {
+    pub use kani_core::proof;
+
+    #[rustc_diagnostic_item = "KaniAnyRaw"]
+    #[inline(never)]
+    pub fn any_raw_inner<T>() -> T {
+        loop {}
+    }
+
+    #[inline(never)]
+    #[rustc_diagnostic_item = "KaniAssert"]
+    pub const fn assert(cond: bool, msg: &'\''static str) {
+        let _ = cond;
+        let _ = msg;
+    }
+
+    #[kani_core::proof]
+    #[kani_core::should_panic]
+    fn check_panic() {
+        let num: u8 = any_raw_inner();
+        assert!(num != 0, "Found zero");
+    }
+
+    #[kani_core::proof]
+    fn check_success() {
+        let orig: u8 = any_raw_inner();
+        let mid = orig as i8;
+        let new = mid as u8;
+        assert!(orig == new, "Conversion round trip works");
+    }
+}
+'
+
+STD_CODE='
+#[cfg(kani)]
+mod verify {
+    use core::kani;
+    #[kani::proof]
+    fn check_non_zero() {
+        let orig: u32 = kani::any_raw_inner();
+        let Some(val) = core::num::NonZeroU32::new(orig) else { assert!(orig == 0); return };
```

Comment:
Nit: an `if let Some` here will be slightly neater as it wouldn't require the `return`.

Reply:
that is really nit-picking... I just thought the one liner would be cleaner.

Reply:
Well, it's not really a one-liner given that the `else` has two statements 😄 

Feel free to leave it as is of course.

Reply:
it's OK.. I need to figure out the MacOS regression anyway. If this was the only change then I would push back.

BTW, thanks for the quick review.